app/app/src/main/kotlin/com/hedvig/app/authenticate/BankIdLoginViewModel.kt

```diff
@@ -68,12 +69,20 @@ class BankIdLoginViewModel(
 
       when (result) {
         is AuthAttemptResult.BankIdProperties -> bankIdProperties.update { result }
-        is AuthAttemptResult.Error -> e { "Got Error when signing in with BankId ${result.message}" }
-        is AuthAttemptResult.ZignSecProperties -> e { "Got ZignSec properties when signing in with BankId" }
-        is AuthAttemptResult.OtpProperties -> e { "Got Otp properties when signing in with BankId" }
-      }
-      if (result !is AuthAttemptResult.BankIdProperties) {
-        startLoginAttemptFailed.update { true }
+        is AuthAttemptResult.Error -> {
+          e { "Got Error when signing in with BankId ${result.message}" }
+          startLoginAttemptFailed.update { "Got Error when signing in with BankId ${result.message}" }
+        }
+
+        is AuthAttemptResult.ZignSecProperties -> {
+          e { "Got ZignSec properties when signing in with BankId" }
+          startLoginAttemptFailed.update { "Got ZignSec properties when signing in with BankId" }
+        }
+
+        is AuthAttemptResult.OtpProperties -> {
+          e { "Got Otp properties when signing in with BankId" }
+          startLoginAttemptFailed.update { "Got Otp properties when signing in with BankId" }
```

Comment:
Should we use a generic error message for the view state here?

Reply:
I see. The error is not a network error though, so it will be confusing for us if members report that they get network errors but in reality there is something wrong with our systems. Its not pretty having hard coded error messages like this, but it would make identifying member reports so easy - we would immediately know what is wrong without having to dig into logs, check members in hope, etc. 

What do you think?

Reply:
Hmm yeah I see what you mean.
The only slow thing with this would be that we need to localize the error message whatever that may be.
We could in this case instead have a message that says something like "There's something wrong on our end", and in the logs make sure to log which one of the alternatives has happened, so we can look into it.
That's at least as you say going to help with not mistakenly making them think it's a network error

Reply:
A localized message would be nicer for the member, on the other hand this error "should never happen". I think in edge cases like this the benefit of knowing exactly what went wrong, directly from the member reporting the error, outweighs having a good UX with member readable error messages.

Reply:
We dont have to do this for all features, but we have had quite a lot of trouble with specifically auth.